6.2 release notes (#10432)
IQSS/dataverse: doc/release-notes/6.2-release-notes.md

# Dataverse 6.2

Please note: To read these instructions in full, please go to https://github.com/IQSS/dataverse/releases/tag/v6.2 rather than the list of releases, which will cut them off.

This release brings new features, enhancements, and bug fixes to the Dataverse software.
Thank you to all of the community members who contributed code, suggestions, bug reports, and other assistance across the project.

## 💡Release highlights

### Return to Author Now Requires a Reason

The Popup for returning to author now requires a reason that will be sent by email to the author.

> Please note that you can still type a creative and meaningful comment such as "The author would like to modify his dataset", "Files are missing", "Nothing to report" or "A curation report with comments and suggestions/instructions will follow in another email" that suits your situation.

Reviewer: I agree, here is my suggestion :
```
### When returning datasets to authors, reviewers can add a note to the author.

The Popup for returning to author now allows to type in a message to explain the reasons of return and potential edits needed, that will be sent by email to the author.

Please note that this note is mandatory, but that you can still type a creative and meaningful comment such as "The author would like to modify his dataset", "Files are missing", "Nothing to report" or "A curation report with comments and suggestions/instructions will follow in another email" that suits your situation.
```

Tiers not specified in this setting will default to `-1` (No Limit). I.e., -d "10000" is equivalent to -d "10000,-1,-1,..."

```
curl http://localhost:8080/api/admin/settings/:RateLimitingCapacityByTierAndAction -X PUT -d '[{
    "tier": 0,
    "limitPerHour": 10,
    "actions": [
        "GetLatestPublishedDatasetVersionCommand",
        "GetPrivateUrlCommand",
        "GetDatasetCommand",
        "GetLatestAccessibleDatasetVersionCommand"
    ]
},
{
    "tier": 0,
    "limitPerHour": 1,
    "actions": [
        "CreateGuestbookResponseCommand",
        "UpdateDatasetVersionCommand",
        "DestroyDatasetCommand",
        "DeleteDataFileCommand",
        "FinalizeDatasetPublicationCommand",
        "PublishDatasetCommand"
    ]
},
{
    "tier": 1,
    "limitPerHour": 30,
    "actions": [
        "CreateGuestbookResponseCommand",
        "GetLatestPublishedDatasetVersionCommand",
        "GetPrivateUrlCommand",
        "GetDatasetCommand",
        "GetLatestAccessibleDatasetVersionCommand",
        "UpdateDatasetVersionCommand",
        "DestroyDatasetCommand",
        "DeleteDataFileCommand",
        "FinalizeDatasetPublicationCommand",
        "PublishDatasetCommand"
    ]
}]'

> ```

Author: So we should just remove it? I am not sure if this is is another place this PR was from @stevenwinship

For more information, see [#10360](https://github.com/IQSS/dataverse/issues/10360).

### Optional Croissant 🥐 Exporter Support

When a Dataverse installation is configured to use a metadata exporter for the [Croissant](https://github.com/mlcommons/croissant) format, the content of the JSON-LD in the **&lt;head>** of dataset landing pages will be replaced with that format. However, both JSON-LD and Croissant will still be available for download from the dataset page and API.

### Search by License

A new search facet called "License" has been added and will be displayed as long as there is more than one license in datasets and datafiles in browse/search results. This facet allow you to filter by license such as CC0, etc.

Also, the Search API now handles license filtering using the `fq` parameter, for example : `/api/search?q=*&fq=license%3A%22CC0+1.0%22` for CC0 1.0. 

For more information, see [#10204](https://github.com/IQSS/dataverse/issues/10204).

### Add .QPJ and .QMD Extensions to Shapefile Handling

> - Support for **.qpj** and **.qmd** files in shapefile uploads has been introduced, ensuring that these files are properly recognized and handled as part of geospatial datasets in Dataverse.

Reviewer: @DS-INRA ToDo

### Ingested Tabular Data Files Can Be Stored Without the Variable Name Header

> Tabular Data Ingest can now save the generated archival files with the list of variable names added as the first tab-delimited line. As the most significant effect of this feature.

Reviewer 2: This is still an incomplete sentence.

> Access API will be able to take advantage of Direct Download for tab. files saved with these headers on S3 - since they no longer have to be generated and added to the streamed content on the fly.

Reviewer 2: I think we mean .tab

An API for converting existing legacy tabular files will be added separately. [this line will need to be changed if we have time to add said API before 6.2 is released]. [TODO]

### Uningest/Reingest Options Available in the File Page Edit Menu

New Uningest/Reingest options are available in the File Page Edit menu. Ingest errors can be cleared by users who can published the associated dataset and by superusers, allowing for a successful ingest to be undone or retried (e.g. after a Dataverse version update or if ingest size limits are changed).

The /api/files/<id>/uningest api also now allows users who can publish the dataset to undo an ingest failure.

## 🪲 Bug fixes

### Publication Status Facet Restored

In version 6.1, the publication status facet location was unintentionally moved to the bottom. In this version, we have restored the original order.

> ### Permissions Required To Assign a Role Have Been Fixed

Reviewer: Not super clear in my opinion, here is a suggestion :
`### Assign a Role With Higher Permissions Than Its Own Role Has Been Fixed`

The permissions required to assign a role have been fixed. It is no longer possible to assign a role that includes permissions that the assigning user doesn't have.

### Geospatial Metadata Block Fields for North and South Renamed

The Geospatial metadata block fields for north and south were labeled incorrectly as ‘Longitudes,’ as reported on #5645. After updating to this version of Dataverse, users will need to update all the endpoints that used ‘northLongitude’ and ‘southLongitude’ to ‘northLatitude’ and ‘southLatitude,’ respectively.

TODO: Whoever puts the release notes together should make sure there is the standard note about updating the schema after upgrading.

### OAI-PMH Error Handling Has Been Improved

OAI-PMH error handling has been improved to display a machine-readable error in XML rather than a 500 error with no further information.

> - /oai?foo=bar will show "No argument 'verb' found"
> - /oai?verb=foo&verb=bar will show "Verb must be singular, given: '[foo, bar]'"

## 💾 Persistence

### Missing Database Constraints

> This release adds two missing database constraints that will assure that the externalvocabularyvalue table only has one entry for each uri and that the oaiset table only has one set for each spec. (In the very unlikely case that your existing database has duplicate entries now, install would fail. This can be checked by running

Reviewer 2: We should still finish this sentence. 😄

### Postgres Versions

This release adds install script support for the new permissions model in Postgres versions 15+, and bumps FlyWay to support Postgres 16.

Postgres 13 remains the version used with automated testing.

## 🌐 API 

### Listing Collection/Dataverse API

Listing collection/dataverse role assignments via API still requires ManageDataversePermissions, but listing dataset role assignments via API now requires only ManageDatasetPermissions.

### New API Endpoint for Clearing an Individual Dataset From Solr

A new Index API endpoint has been added allowing an admin to clear an individual dataset from Solr.

### New Accounts Metrics API

Users can retrieve new types of metrics related to user accounts. The new capabilities are [described](https://guides.dataverse.org/en/6.2/api/metrics.html) in the guides.

### New canDownloadAtLeastOneFile Endpoint

The `/api/datasets/{id}/versions/{versionId}/canDownloadAtLeastOneFile` endpoint has been created.

This API endpoint indicates if the calling user can download at least one file from a dataset version. Note that Shibboleth group permissions are not considered.

> ### Harvesting Client Endpoint Extended

Reviewer: I think it will be a little bit hidden in the API section as it is not really related too :wink:

> ### Harvesting Client Endpoint Extended

Reviewer: maybe a mention in the features with a reference to this section would be enough ?

### Endpoint Fixed: Datasets Metadata

The API endpoint `api/datasets/{id}/metadata` has been changed to default to the latest version of the dataset that the user has access.

## 📖 Guides

### New QA Guide

A new QA Guide is intended mostly for the core development team but may be of interest to contributors.

### Container Guide, Documentation for Faster Redeploy

In the Container Guide, documentation for developers on how to quickly redeploy code has been added for Netbeans and improved for IntelliJ.

Also in the context of containers, a new option to skip deployment has been added and the war file is now consistently named "dataverse.war" rather than having a version in the filename, such as "dataverse-6.1.war". This predictability makes tooling easier.

Finally, an option to create tabs in the guides using [Sphinx Tabs](https://sphinx-tabs.readthedocs.io) has been added. (You can see the tabs in action in the "dev usage" page of the Container Guide.) To continue building the guides, you will need to install this new dependency by re-running  `pip install -r requirements.txt`.

> ### Evaluation Version Tutorial on the Containers Guide

Reviewer 2: I'm fine with the dev stuff (faster redeploy, QA Guide) being relegated to a section on guides, but the container demo tutorial has a much wider audience. I still think it should be elevated to be a highlighted feature.

> ### Evaluation Version Tutorial on the Containers Guide

Reviewer 2: JP and I talked about it. We'll leave the demo tutorial where it is.

> The Container Guide now containers a tutorial for running Dataverse in containers for demo or evaluation purposes: https://guides.dataverse.org/en/6.2/container

Reviewer 2: I'm just leaving this not on the last line of the release notes but we need the rest of it. 😄 

We especially need the upgrade instructions. That's the hardest part.

# Dataverse 6.2

Please note: To read these instructions in full, please go to https://github.com/IQSS/dataverse/releases/tag/v6.2 rather than the list of releases, which will cut them off.

This release brings new features, enhancements, and bug fixes to the Dataverse software.
Thank you to all of the community members who contributed code, suggestions, bug reports, and other assistance across the project.

## 💡Release Highlights

### Search and Facet by License  

A new search facet called "License" has been added and will be displayed as long as there is more than one license in datasets and datafiles in browse/search results. This facet allow you to filter by license such as CC0, etc.

Also, the Search API now handles license filtering using the `fq` parameter, for example : `/api/search?q=*&fq=license%3A%22CC0+1.0%22` for CC0 1.0. 

For more information, see [#10204](https://github.com/IQSS/dataverse/issues/10204).

> ### Return to Author Now Requires a Reason

Reviewer 2: ```suggestion
### When Returning Datasets to Authors, Reviewers Can Add a Note to the Author
```

(suggested by @DS-INRA at https://github.com/IQSS/dataverse/pull/10432#discussion_r1541094927 )

The Popup for returning to author now allows to type in a message to explain the reasons of return and potential edits needed, that will be sent by email to the author.

Please note that this note is mandatory, but that you can still type a creative and meaningful comment such as "The author would like to modify his dataset", "Files are missing", "Nothing to report" or "A curation report with comments and suggestions/instructions will follow in another email" that suits your situation.

<a id="multiple-pid-sup"></a>
### Support for Using Multiple PID Providers

This release adds support for using multiple PID (DOI, Handle, PermaLink) providers, multiple PID provider accounts

> (managing a given protocol, authority,separator, shoulder combination), assigning PID provider accounts to specific collections,

Reviewer 2: ```suggestion
(managing a given protocol, authority, separator, shoulder combination), assigning PID provider accounts to specific collections,
```

and supporting transferred PIDs (where a PID is managed by an account when its authority, separator, and/or shoulder don't match
the combination where the account can mint new PIDs). It also adds the ability for additional provider services beyond the existing
DataCite, EZId, Handle, and PermaLink providers to be dynamically added as separate jar files.

These changes require per-provider settings rather than the global PID settings previously supported. While backward compatibility 
for installations using a single PID Provider account is provided, updating to use the new microprofile settings is highly recommended
and will be required in a future version.

[New microprofile settings](#microprofile-settings)

> 

Reviewer 2: Can we link to the updated guides section on configuring PIDs?

Author: There is one setting from there that doesn't belong to the same PR but let me look at this.

Author: Is this the right link? https://dataverse-guide--10432.org.readthedocs.build/en/10432/installation/config.html#global-settings

> ### Rate Limiting Using JCache (With Hazelcast As Provided by Payara)

Reviewer 2: ```suggestion
### Rate Limiting
```

The details of JCache, etc. probably aren't of interest.

The option to rate limit has been added to prevent users from over taxing the system either deliberately or by runaway automated processes.
Rate limiting can be configured on a tier level with tier 0 being reserved for guest users and tiers 1-any for authenticated users.
Superuser accounts are exempt from rate limiting.
Rate limits can be imposed on command APIs by configuring the tier, the command, and the hourly limit in the database.
Two database settings configure the rate limiting.
Note: If either of these settings exist in the database rate limiting will be enabled.
If neither setting exists rate limiting is disabled.

> `:RateLimitingDefaultCapacityTiers` is a comma separated list of default values for each tier.
> In the following example, the default for tier `0` (guest users) is set to 10,000 calls per command per hour and tier `1` (authenticated users) is set to 20,000 calls per command per hour.
> Tiers not specified in this setting will default to `-1` (No Limit). I.e., -d "10000" is equivalent to -d "10000,-1,-1,..."
> 
> For more details check the detailed guide on [this link](https://guides.dataverse.org/en/6.2/installation/config.html#rate-limiting).
> 
> Hazelcast is configured in Payara and should not need any changes for this feature

Reviewer 2: ```suggestion
For more information check the detailed guide at [this link](https://guides.dataverse.org/en/6.2/installation/config.html#rate-limiting).
```

I think this is enough. However, let's make sure both new database settings are mentioned in the release notes.

> ### Simplified SMTP configuration

Reviewer 2: ```suggestion
### Simplified SMTP Configuration
```

(seems like we're going for Title Case)

This behavior is controlled by the new setting **:StoreIngestedTabularFilesWithVarHeaders**. It is false by default, preserving the legacy behavior. When enabled, Dataverse will be able to handle both the newly ingested files, and any already-existing legacy files stored without these headers transparently to the user. E.g. the access API will continue delivering tab-delimited files **with** this header line, whether it needs to add it dynamically for the legacy files, or reading complete files directly from storage for the ones stored with it.

We are planning to add an API for converting existing legacy tabular files in a future release.

### Uningest/Reingest Options Available in the File Page Edit Menu

New Uningest/Reingest options are available in the File Page Edit menu. Ingest errors can be cleared by users who can published the associated dataset and by superusers, allowing for a successful ingest to be undone or retried (e.g. after a Dataverse version update or if ingest size limits are changed).

The /api/files/<id>/uningest api also now allows users who can publish the dataset to undo an ingest failure.

> ### Sphinx Guides now Support Markdown Format and Tabs

Reviewer 2: ```suggestion
### Sphinx Guides Now Support Markdown Format and Tabs
```

```
## 🪲 Bug fixes

### Publication Status Facet Restored

In version 6.1, the publication status facet location was unintentionally moved to the bottom. In this version, we have restored the original order.

### Assign a Role With Higher Permissions Than Its Own Role Has Been Fixed

The permissions required to assign a role have been fixed. It is no longer possible to assign a role that includes permissions that the assigning user doesn't have.

### Geospatial Metadata Block Fields for North and South Renamed

The Geospatial metadata block fields for north and south were labeled incorrectly as ‘Longitudes,’ as reported on #5645. After updating to this version of Dataverse, users will need to update all the endpoints that used ‘northLongitude’ and ‘southLongitude’ to ‘northLatitude’ and ‘southLatitude,’ respectively.

### OAI-PMH Error Handling Has Been Improved

OAI-PMH error handling has been improved to display a machine-readable error in XML rather than a 500 error with no further information.

> - /oai?foo=bar will show "No argument 'verb' found"
> - /oai?verb=foo&verb=bar will show "Verb must be singular, given: '[foo, bar]'"

### Granting File Access Without Access Request

> A bug introduced with the guestboook-at-request, requests are not deleted when granted, they are now given the state granted.

Reviewer 2: ```suggestion
A bug introduced with the guestbook-at-request, requests are not deleted when granted, they are now given the state granted.
```

```
SELECT uri, count(*) FROM externalvocabularyvalue group by uri;
```
And:
```
SELECT spec, count(*) FROM oaiset group by spec;
```
Then removing any duplicate rows (where count>1).

### Universe Field in Variablemetadata Table Changed

Universe field in variablemetadata table was changed from **varchar(255)** to **text**. The change was made to support longer strings in "universe" metadata field, similar to the rest of text fields in variablemetadata table.

### Postgres Versions

This release adds install script support for the new permissions model in Postgres versions 15+, and bumps FlyWay to support Postgres 16.

Postgres 13 remains the version used with automated testing.

## 🌐 API 

### Listing Collection/Dataverse API

Listing collection/dataverse role assignments via API still requires ManageDataversePermissions, but listing dataset role assignments via API now requires only ManageDatasetPermissions.

### New API Endpoint for Clearing an Individual Dataset From Solr

> A new Index API endpoint has been added allowing an admin to clear an individual dataset from Solr.

Reviewer 2: How? Where? Can we link to the right place in the guides?

### New Accounts Metrics API

> Users can retrieve new types of metrics related to user accounts. The new capabilities are [described](https://guides.dataverse.org/en/6.2/api/metrics.html) in the guides.

Reviewer 2: It would be nice to link to the specific section on user accounts.

Author: It seems it is a parameter so all the guide would be related to this?

### New canDownloadAtLeastOneFile Endpoint

The `/api/datasets/{id}/versions/{versionId}/canDownloadAtLeastOneFile` endpoint has been created.

This API endpoint indicates if the calling user can download at least one file from a dataset version. Note that Shibboleth group permissions are not considered.

### Harvesting Client Endpoint Extended

The API endpoint `api/harvest/clients/{harvestingClientNickname}` has been extended to include the following fields:

> - **allowHarvestingMissingCVV**: enable/disable allowing datasets to be harvested with Controlled Vocabulary Values that existed in the originating Dataverse Project but are not in the harvesting Dataverse Project. Default is false.

Reviewer 2: ```suggestion
- **allowHarvestingMissingCVV**: enable/disable allowing datasets to be harvested with controlled vocabulary values that exist in the originating Dataverse server but are not present in the harvesting Dataverse server. The default is false.
```

> *Note: This setting is only available to the API and not currently accessible/settable via the UI*

Reviewer 2: ```suggestion
*Note: This setting is only available to the API and not currently accessible/settable via the UI.*
```

### Version Files Endpoint Extended

The response for getVersionFiles `/api/datasets/{id}/versions/{versionId}/files` endpoint has been modified to include a total count of records available **totalCount:x**.
This will aid in pagination by allowing the caller to know how many pages can be iterated through. The existing API (getVersionFileCounts) to return the count will still be available.

### Metadata Blocks Endpoint Extended

The API endpoint `/api/metadatablocks/{block_id}` has been extended to include the following fields:

- **isRequired**: Whether or not this field is required
- **displayOrder**: The display order of the field in create/edit forms
- **typeClass**: The type class of this field ("controlledVocabulary", "compound", or "primitive")

> ### Get File Citation As JSON

Reviewer 2: ```suggestion
### Get File Citation as JSON
```

It is now possible to retrieve via API the file citation as it appears on the file landing page. It is formatted in HTML and encoded in JSON.

> 

Reviewer 2: It would be nice to link to the guides on this.

A new endpoint `api/files/{id}/versions/{datasetVersionId}` has been created. This endpoint returns the file metadata present in the requested dataset version. To specify the dataset version, you can use `:latest-published`, `:latest`, `:draft` or `1.0` or any other available version identifier.

The endpoint supports the *includeDeaccessioned* and *returnDatasetVersion* optional query parameters, as does the `api/files/{id}` endpoint.

`api/files/{id}/draft` endpoint is no longer available in favor of the new endpoint `api/files/{id}/versions/{datasetVersionId}`, which can use the version identifier ``:draft`` (`api/files/{id}/versions/:draft`) to obtain the same result.

### Datasets, Dataverse Collections, and Datafiles Endpoints Extended

The API endpoints for getting datasets, Dataverse collections, and datafiles have been extended to support the following optional 'returnOwners' query parameter.

Including the parameter and setting it to true will add a hierarchy showing which dataset and dataverse collection(s) the object is part of to the json object returned. 

> 

Reviewer 2: It would be nice to add links to the guides here.

### Endpoint Fixed: Datasets Metadata

The API endpoint `api/datasets/{id}/metadata` has been changed to default to the latest version of the dataset that the user has access.

### Experimental Make Data Count processingState API

An experimental Make Data Count processingState API has been added. For now it has been documented in the (developer guide)[https://guides.dataverse.org/en/6.2/developers/make-data-count.html#processing-archived-logs].

> 

Reviewer 2: At the end of the API section, should we say, "For a full list of backward-incompatible API changes, please see the "breaking changes" section below"?

If you are running Payara as a non-root user (and you should be!), **remember not to execute the commands below as root**. Use `sudo` to change to that user first. For example, `sudo -i -u dataverse` if `dataverse` is your dedicated application user.

In the following commands we assume that Payara 6 is installed in `/usr/local/payara6`. If not, adjust as needed.

`export PAYARA=/usr/local/payara6`

(or `setenv PAYARA /usr/local/payara6` if you are using a `csh`-like shell)

1\. Usually, when a Solr schema update is released, we recommend deploying the new version of Dataverse, then updating the `schema.xml` on the solr side. With 6.2, we recommend to install the base schema first. Without it Dataverse 6.2 is not going to be able to show any results after the initial deployment. If your instance is using any custom metadata blocks, you will need to further modify the schema, see the laset step of this instruction (step 8).

> - Stop Solr instance (usually `service solr stop`, depending on Solr installation/OS, see the [Installation Guide](https://guides.dataverse.org/en/5.14/installation/prerequisites.html#solr-init-script))

Reviewer 2: ```suggestion
- Stop Solr instance (usually `service solr stop`, depending on Solr installation/OS, see the [Installation Guide](https://guides.dataverse.org/en/6.2/installation/prerequisites.html#solr-init-script))
```

- Replace schema.xml

  - `cd /usr/local/solr/solr-9.3.0/server/solr/collection1/conf`
  - `wget https://raw.githubusercontent.com/IQSS/dataverse/master/conf/solr/9.3.0/schema.xml`

- Start Solr instance (usually `service solr start`, depending on Solr/OS)

2\. Undeploy the previous version.

> - `$PAYARA/bin/asadmin undeploy dataverse-6.0`

Reviewer 2: ```suggestion
- `$PAYARA/bin/asadmin undeploy dataverse-6.1`
```

3\. Stop Payara and remove the generated directory

- `service payara stop`
- `rm -rf $PAYARA/glassfish/domains/domain1/generated`

4\. Start Payara

- `service payara start`

5\. Deploy this version.

> - `$PAYARA/bin/asadmin deploy dataverse-6.1.war`

Reviewer 2: ```suggestion
- `$PAYARA/bin/asadmin deploy dataverse-6.2.war`
```

wget https://github.com/IQSS/dataverse/releases/download/v6.2/biomedical.tsv

curl http://localhost:8080/api/admin/datasetfield/load -H "Content-type: text/tab-separated-values" -X POST --upload-file scripts/api/data/metadatablocks/biomedical.tsv
```

8\. For installations with custom or experimental metadata blocks:

- Stop Solr instance (usually `service solr stop`, depending on Solr installation/OS, see the [Installation Guide](https://guides.dataverse.org/en/6.2/installation/prerequisites.html#solr-init-script))

- Run the `update-fields.sh` script that we supply, as in the example below (modify the command lines as needed to reflect the correct path of your solr installation):
```
	wget https://raw.githubusercontent.com/IQSS/dataverse/master/conf/solr/9.3.0/update-fields.sh
	chmod +x update-fields.sh
	curl "http://localhost:8080/api/admin/index/solr/schema" | ./update-fields.sh /usr/local/solr/solr-9.3.0/server/solr/collection1/conf/schema.xml
```
     
- Restart Solr instance (usually `service solr restart` depending on solr/OS)

9\. Reindex Solr:

>    For details, see https://guides.dataverse.org/en/6.0/admin/solr-search-index.html but here is the reindex command:

Reviewer 2: ```suggestion
   For details, see https://guides.dataverse.org/en/6.2/admin/solr-search-index.html but here is the reindex command:
```

```
   curl http://localhost:8080/api/admin/index
```

***

## ⚙️ New Settings 

> ### Microprofile settings

Reviewer 2: ```suggestion
### MicroProfile Settings
```

> - dataverse.pid.*.excluded-list
> - dataverse.pid.*.datacite.mds-api-url
> - dataverse.pid.*.datacite.rest-api-url
> - dataverse.pid.*.datacite.username
> - dataverse.pid.*.datacite.password
> - dataverse.pid.*.ezid.api-url
> - dataverse.pid.*.ezid.username
> - dataverse.pid.*.ezid.password
> - dataverse.pid.*.permalink.base-url
> - dataverse.pid.*.permalink.separator
> - dataverse.pid.*.handlenet.index
> - dataverse.pid.*.handlenet.independent-service
> - dataverse.pid.*.handlenet.auth-handle
> - dataverse.pid.*.handlenet.key.path
> - dataverse.pid.*.handlenet.key.passphrase
> - dataverse.spi.pidproviders.directory
> - dataverse.solr.concurrency.max-async-indexes

[⬅️ Go back](#multiple-pid-sup)

## SMTP Settings: 

> - dataverse.mail.system-email
> - dataverse.mail.mta.host
> - dataverse.mail.mta.port
> - dataverse.mail.mta.ssl.enable
> - dataverse.mail.mta.auth
> - dataverse.mail.mta.user
> - dataverse.mail.mta.password
> - dataverse.mail.mta.allow-utf8-addresses
> - Plus many more for advanced usage and special provider requirements. See [configuration guide for a full list](https://guides.dataverse.org/en/6.2/installation/config.html#dataverse-mail-mta).

> [⬅️ Go back](#simplified-smtp-configuration)

Reviewer 2: Can we list the new database settings as well?

Author: What would be the settings to list here? I couldn't find something on the note and was not sure what to grab from the PR for this. 😄

Reviewer 2: You can see "New JVM Options and DB Settings" at https://github.com/IQSS/dataverse/releases/tag/v5.10 for an example of a release where both JVM/MPCONFIG and database settings were added.
